src/lib/protocols/dns.c

```diff
@@ -513,8 +517,30 @@ static void ndpi_search_dns(struct ndpi_detection_module_struct *ndpi_struct, st
 
   if(flow->packet_counter > 3)
     NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    
+  if((flow->packet.detected_protocol_stack[0] == NDPI_PROTOCOL_DNS)
+	 || (flow->packet.detected_protocol_stack[1] == NDPI_PROTOCOL_DNS)) {
+	   
+  	if(flow->packet.udp != NULL && flow->packet.payload_packet_len > PKT_LEN_ALERT)
```

Comment:
Could you add a comment about EDNS above this check, please? Something like "TODO: add supporto to RFC6891 to avoid some false positives"